From 0f8f207bc00ce7d7a7e5b516bef17c424ed5b2ab Mon Sep 17 00:00:00 2001
From: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
Date: Tue, 25 Feb 2020 23:21:46 -0700
Subject: [PATCH 1242/1851] mtd: rawnand: pl353: Increase the data in/out
 lengths

commit 79801ede37c4275ce923951c54438b14bdd64538 from
https://github.com/Xilinx/linux-xlnx.git

There are cases where the nand transfers can have a length of more than
page size. which is page size + oob length. Hence update the data in/out
lengths from 2048 to 2112(2048 + 64).

Signed-off-by: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/mtd/nand/raw/pl353_nand.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/raw/pl353_nand.c b/drivers/mtd/nand/raw/pl353_nand.c
index 1f175b26724c..1e5b05acdab3 100644
--- a/drivers/mtd/nand/raw/pl353_nand.c
+++ b/drivers/mtd/nand/raw/pl353_nand.c
@@ -58,6 +58,7 @@
 #define PL353_TREA_MAX_VALUE		1
 #define PL353_MAX_ECC_CHUNKS		4
 #define PL353_MAX_ECC_BYTES		3
+#define PL353_MAX_CHUNK_SIZE		2112
 
 struct pl353_nfc_op {
 	u32 cmnds[2];
@@ -905,11 +906,11 @@ static const struct nand_op_parser pl353_nfc_op_parser = NAND_OP_PARSER(
 		NAND_OP_PARSER_PAT_ADDR_ELEM(true, 7),
 		NAND_OP_PARSER_PAT_CMD_ELEM(true),
 		NAND_OP_PARSER_PAT_WAITRDY_ELEM(true),
-		NAND_OP_PARSER_PAT_DATA_IN_ELEM(true, 2048)),
+		NAND_OP_PARSER_PAT_DATA_IN_ELEM(true, PL353_MAX_CHUNK_SIZE)),
 	NAND_OP_PARSER_PATTERN(pl353_nand_exec_op_cmd,
 		NAND_OP_PARSER_PAT_CMD_ELEM(true),
 		NAND_OP_PARSER_PAT_ADDR_ELEM(true, 7),
-		NAND_OP_PARSER_PAT_DATA_OUT_ELEM(true, 2048),
+		NAND_OP_PARSER_PAT_DATA_OUT_ELEM(true, PL353_MAX_CHUNK_SIZE),
 		NAND_OP_PARSER_PAT_CMD_ELEM(true),
 		NAND_OP_PARSER_PAT_WAITRDY_ELEM(true)),
 	);
-- 
2.31.1

